src/components/user/user.edgedb.repository.ts

```diff
@@ -46,6 +52,20 @@ export class UserEdgeDBRepository
     return !!result;
   }
 
+  getUserByEmailAddress(
```

Comment:
@CarsonF - There's an opportunity here to remove `doesEmailAddressExist` and replace with this new query since it could be used as a dual purpose.  I believe if this happened though, I'd have to remove the `e.assert_exists` and handle the `null` back in the service layer with an exception.  Would that work?  Is that even desirable?  Or is there another way?  I suppose we could also leave both methods as is...

Reply:
I would yes. I would make the query nullable. There's not an expectation that the email will exist IMO. Contrasted to IDs which we expect to exist.

Reply:
I would agree on the removal, it doesn't seem as if there's anything `doesEmailAddressExist` is accomplishing that a `null` from your new method couldn't handle.

Reply:
Yeah I'm saying the not finding a user with this email is expected behavior. The caller can do whatever they want with the null.